### ISSUE TITLE
Support for multiple CodeArtifact repositories?

### ISSUE THREAD

**Issue by jfly** [jmkeyes/keyrings.codeartifact#2]
The [current configuration options](https://github.com/jmkeyes/keyrings.codeartifact/tree/1.2.0?tab=readme-ov-file#config) don't seem to be nuanced enough to support multiple CodeArtifact repositories at once.

It looks like this would be a straightforward change ([at the time we load the config](https://github.com/jmkeyes/keyrings.codeartifact/blob/1.2.0/keyrings/codeartifact.py#L89), we already know the url of the repository we're dealing with).

Would you be open to adding more granularly scoped configuration options? I'm not terribly familiar with ini files, but one way of doing this would be with more sections:

```ini
$ cat keyringrc.cfg
[codeartifact]
# 30 minutes
token_duration = 1800
profile_name = global-profile

[codeartifact.domain1-dev]
url = "https://domain1-999999999999.d.codeartifact.us-west-2.amazonaws.com/pypi/domain1-dev/simple/"
profile_name = domain1-dev

[codeartifact.domain1-prod]
url = "https://domain1-999999999999.d.codeartifact.us-west-2.amazonaws.com/pypi/domain1-prod/simple/"
profile_name = domain1-prod
```

In the above example, the names of the sections don't really matter, they just need to be unique. An alternative would be to jam these urls into the section names (for example `[codeartifact.https://domain1-999999999999.d.codeartifact.us-west-2.amazonaws.com/pypi/domain1-prod/simple/]`), but that feels a little ugly/unwieldy IMO.

This would be backwards compatible with the existing options, but would take precedence if the url matches one of these subsections.

If you're open to this, I'd be happy to send in a PR implementing it.

(tangentially related: https://github.com/jaraco/keyring/issues/682)

**Comment by jmkeyes** [comment]
Hey @jfly, thanks for submitting this issue! 

I've pushed up an implementation of this to the [`multiple-config-sections`](/jmkeyes/keyrings.codeartifact/tree/multiple-config-sections) branch. Here's a sample from the [`README`](/jmkeyes/keyrings.codeartifact/tree/multiple-config-sections?tab=readme-ov-file#multiple-section-configuration-experimental):

> ```ini
> # These are the defaults for other sections.
> [codeartifact]
> token_duration=1800
> 
> # Use for any repositories in account 000000000000.
> [codeartifact account="000000000000"]
> profile_name=special_profile
>
> # Use for account 999999999999 with a repository named "staging".
> [codeartifact account="999999999999" name="staging"]
> profile_name=staging_profile
> ```

If you'd like to give it a try yourself, you can install it with this `pip` command:

```
# Or add the following URL to your `requirements.txt`. 
python -m pip install git+https://github.com/jmkeyes/keyrings.codeartifact.git@multiple-config-sections
```

Let me know what you think!

**Comment by jmkeyes** [comment]
I've added some unit testing around this functionality and released it as 1.3.0 in 46e83fe0. 

Let me know if it works for you!

**Comment by jfly** [comment]
Thanks, @jmkeyes!
